stock.py

```diff
@@ -104,3 +104,11 @@ class Move(metaclass=PoolMeta):
     def __setup__(cls):
         super(Move, cls).__setup__()
         cls.lot.context['from_move'] = Eval('id')
+
+    def get_cost_price(self, product_cost_price=None):
+        cost_price = super().get_cost_price(product_cost_price=None)
+
+        with Transaction().set_context(date=self.effective_date):
+            if self.lot and self.lot.cost_price is not None and self.cost_price is None:
```

Comment:
I would change the method like this:

```
 def get_cost_price(self, product_cost_price=None):
        pool = Pool()
        Lot = pool.get('stock.lot')

        cost_price = super().get_cost_price(product_cost_price=None)

        if self.lot and self.to_location.type == 'production':
            with Transaction().set_context(date=self.effective_date):
                lot = Lot(self.cost.id) # Need to reinstantiate to ensure the context is correct
                if  lot.cost_price is not None:
                    return lot.cost_price

    return cost_price
```

Note that I check for the `to_location`but also I do not take into account if `self.cost_price is None` which it was being checked and I don't know why. I also ensure we reinstantiate so the context correct context is used.

Reply:
>>> lot = Lot(self.cost.id) # Need to reinstantiate to ensure the context is correct

Cal reinstanciar el lot:

```
Lot(self.lot.id)
```

Procedim a crear un scenario test

Reply:
f6399b7a8f50b18026d6a1dc4afec73972d7e8d3